### ISSUE TITLE
SymbolProc may break code

### ISSUE THREAD

**Issue by klippx** [bbatsov/rubocop#2135]
I have a piece of controller code like this:
```ruby
class Api::SubstancesController < Api::BaseController
  ...
  def destroy
    super([:pharmaceut]) do |substance|
      substance.destroyable?
    end
  end
  ...
end
````
Rubocop wants to autocorrect this to:
```
> rubocop -a app/controllers/api/substances_controller.rb
Inspecting 1 file
E

Offenses:

app/controllers/api/substances_controller.rb:20:5: C: [Corrected] Pass &:destroyable? as an argument to  instead of a block.
    super([:pharmaceut]) do |substance|
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
app/controllers/api/substances_controller.rb:20:25: E: unexpected token tLPAREN2
    super([:pharmaceut])(&:destroyable?)
                        ^
app/controllers/api/substances_controller.rb:42:1: E: unexpected token $end

1 file inspected, 3 offenses detected, 1 offense corrected
````

Rubocop even detects two Errors - all due to the code it generated on its own.

The final result:

```ruby
  def destroy
    super([:pharmaceut])(&:destroyable?)
  end
````

As you can see this is clearly erroneous syntax. 

Version
````
> rubocop -v
warning: parser/current is loading parser/ruby21, which recognizes
warning: 2.1.6-compliant syntax, but you are running 2.1.5.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
0.33.0
````

I have tried to add 'super' to ignored methods in config (seeing this cop inspects this configuration attribute) but that did not help
```yml
Style/BlockDelimiters:
  IgnoredMethods:
    - super
    - lambda
    - proc
    - it
```

**Comment by bbatsov** [comment]
`super` is not a method. We need to add an exception for it in the code.

**Comment by bbatsov** [comment]
Or even better - we should auto-correct code with `super` as:

```
def destroy
    super([:pharmaceut], &:destroyable?)
end
```

**Comment by klippx** [comment]
Thanks, @bbatsov !
